punjab/jabber.py

```diff
@@ -103,9 +157,12 @@ def associateWithStream(self, xs):
             init.required = required
             xs.initializers.append(init)
 
-    def _reset(self):
+    def _reset(self, shallow=False):
         # need this to be in xmlstream
-        stream = domish.elementStream()
```

Comment:
elementStream returns an ExpatElementStream or an SuxElementStream depending if expat is available or not. If someone is running punjab without that dependency it will fail. 

If you could mimic this function, I think that would be ok. 
https://github.com/twisted/twisted/blob/trunk/src/twisted/words/xish/domish.py#L591